modules/portfolio/lib/portfolio-data.service.ts

```diff
@@ -92,35 +92,50 @@ export class PortfolioDataService {
         const blockNumber = parseInt(block.number);
         const poolShares = [];
 
-        for (const sharesOwned of user.sharesOwned ?? []) {
-            const pool = pools.find((pool) => pool.id === sharesOwned.poolId.id);
-
-            if (!pool) {
-                continue;
+        const poolIds = [];
+        for (const farm of farmUsers) {
+            if (farm.pool) {
+                poolIds.push(farm.pool?.pair);
             }
+        }
+        for (const sharesOwned of user.sharesOwned ?? []) {
+            poolIds.push(sharesOwned.poolId.id);
+        }
 
-            const poolSnapshot: PrismaBalancerPoolSnapshotWithTokens = {
-                ...pools.find((pool) => pool.id === sharesOwned.poolId.id)!,
-                poolId: pool.id,
-                amp: pool.amp ?? null,
-                blockNumber,
-                tokens: this.getPoolTokens(pool, pools).map((token) => ({
-                    ...token,
-                    snapshotId: '',
+        for (const pool of pools) {
+            // pool.address for masterchef subgraph, pool.id for sharesOwned in balancer subgraph
+            if (poolIds.includes(pool.address) || poolIds.includes(pool.id)) {
```

Comment:
I see why you're doing this, but this is pretty confusing. At least rename your `poolIds` variable to something else. The pool Id is something different then the address

Reply:
You're right, it's stupid like this. Introduced a second array for the pool addresses. No need to save space. Better?